alphadia/transferlearning/metrics.py

```diff
@@ -5,67 +5,55 @@
 from peptdeep.model.ms2 import calc_ms2_similarity
 
 
-class MetricAccumulator:
-    """
-    Accumulator for any metric.
-    """
-
-    def __init__(self, name: str):
-        self.name = name
-        self.stats = None
-
-    def accumulate(self, epoch: int, loss: float):
-        """
-        Accumulate a metric at a given epoch.
-
-        Parameters
-        ----------
-        epoch : int
-            The epoch at which the metric was calculated.
-        loss : float
-            The value of the metric.
-
-        """
-
-        new_stats = pd.DataFrame({self.name: [loss]})
-        new_stats.index = [epoch]
-        new_stats.index.name = "epoch"
-        self.stats = (
-            new_stats if self.stats is None else pd.concat([self.stats, new_stats])
-        )
-
-
 class TestMetricBase:
     """
-    An Base class for test metrics. Test metrics are classes that calculate a metric on the test set at a given epoch
-    and accumulate the metric over time for reporting.
+    An Base class for test metrics. Test metrics are classes that calculate a metric on the test set at a given epoch.
     """
 
     def __init__(self, columns: List[str]):
         self.columns = columns  # a list of column names for the stats dataframe
-        self.stats = (
-            None  # Stats is a pandas dataframe that stores the test metric over time
-        )
 
-    def _update_stats(self, new_stats: pd.DataFrame, epoch: int):
+    def _to_long_format(
```

Comment:
Would it also be possible to calculate the metrics immediately as long format without the need to call `_to_long_format()` ? I guess this is due to peptdeep metrics being calculated in dense format?